src/index.js

```diff
@@ -1,47 +1,145 @@
+      </div>
+    `;
     const liTag = document.createElement('li');
-    liTag.classList.add('px-10-py-15');
-    liTag.classList.add('text-gray');
-    liTag.classList.add('border-bottom');
     liTag.innerHTML = htmlTemplate;
     todoList.appendChild(liTag);
   });
+  // Get all more buttons
+  moreBtn = document.querySelectorAll('.drop-btn');
+  moreBtn.forEach((eachMoreBtn, btnIndex) => {
+    // toggle more button to edit or delete a task
+    eachMoreBtn.addEventListener('click', () => {
+      const grandParentElement = eachMoreBtn.parentElement.parentElement;
+      const delBtnContainer = grandParentElement.parentElement.lastElementChild;
+      const checkClassName = delBtnContainer.className.includes('hide');
+      if (checkClassName) {
+        delBtnContainer.classList.remove('hide');
+        delBtnContainer.classList.add('show');
+        grandParentElement.classList.add('hide');
+
+        // Delete a task
+        const taskIndex = delBtnContainer.firstElementChild.id;
+        const deleteTask = delBtnContainer.lastElementChild.firstElementChild;
+        deleteTask.addEventListener('click', () => {
+          task.forEach((tas, index) => {
+            if (parseInt(taskIndex, 10) === index) {
+              task.splice(index, 1);
+              saveTask();
+              loadTask();
+            }
+          });
+        });
+        // Edit a task
+        const updateTask = delBtnContainer.firstElementChild.nextElementSibling;
+        updateTask.addEventListener('blur', () => {
+          if (updateTask.value) {
+            const taskData = {
+              description: '',
+              completed: false,
+            };
+
+            taskData.description = updateTask.value;
+            task[btnIndex] = taskData;
+            updateTask.value = taskData.description;
+            saveTask();
+            loadTask();
+          }
+        });
+      }
+    });
+  });
+
+  check();
 };
 
-taskList();
+// Method for creating new task
+const createTask = () => {
+  if (task.length === 0) {
+    todoTask.id = 1;
+    saveTaskId();
+  } else {
+    todoTask.id += 1;
+    saveTaskId();
+  }
+
+  const taskData = {
+    id: '',
+    description: '',
+    completed: false,
+  };
+
+  taskData.id = todoTask.id;
+  taskData.description = newTask.value;
+  task.push(taskData);
+  saveTask();
+  newTask.value = '';
+  loadTask();
+};
+
+// Add new task
+const addTask = document.getElementById('new-task');
+addTask.addEventListener('keyup', (event) => {
+  if (newTask.value) {
+    if (event.code === 'Enter') {
+      createTask();
+    }
+  }
+});
+
+// Delete completed task
+const delCheckedTask = document.getElementById('clear-task');
+delCheckedTask.addEventListener('click', () => {
+  if (task.length !== 0) {
+    const selected = document.querySelectorAll('[checked=checked]');
+    for (let i = selected.length - 1; i >= 0; i -= 1) {
+      task.splice(selected[i].id, 1);
+    }
+    saveTask();
+    loadTask();
+  }
+});
```

Comment:
- [ ] As requested for this project please 🙏🏽 use the filter() method to clear all completed tasks and ensure that the index update when the tasks are removed.

> Implement a function for the "Clear all completed" button (use filter() method).

![image](https://user-images.githubusercontent.com/58642949/230721877-24ec00fc-61ae-4e0f-b10e-a00eeddfef02.png)